pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1870,3 +1871,719 @@ var _ = Describe("controlplanemachinesetgenerator controller on GCP", func() {
+
+				It("should update, but not duplicate the failure domains on the ControlPlaneMachineSet", func() {
+					Eventually(komega.Object(cpms)).Should(HaveField("Spec.Template.OpenShiftMachineV1Beta1Machine.FailureDomains", Equal(cpms5FailureDomainsBuilderOpenStack.BuildFailureDomains())))
+				})
+			})
+		})
+
+		Context("with state Inactive and up to date", func() {
+			BeforeEach(func() {
+				By("Creating an up to date and Inactive Control Plane Machine Set")
+				// Create an Inactive ControlPlaneMachineSet with a Provider Spec that
+				// match the youngest control plane machine (i.e. it's up to date).
+				cpms = cpmsInactive5FDsBuilderOpenStack.WithNamespace(namespaceName).Build()
+				Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
+			})
+
+			It("should keep the ControlPlaneMachineSet up to date and not change it", func() {
+				cpmsVersion := cpms.ObjectMeta.ResourceVersion
+				Consistently(komega.Object(cpms)).Should(HaveField("ObjectMeta.ResourceVersion", cpmsVersion))
+			})
+
+		})
+
+		Context("with state Active and outdated", func() {
+			BeforeEach(func() {
+				By("Creating an outdated and Active Control Plane Machine Set")
+				// Create an Active ControlPlaneMachineSet with a Provider Spec that
+				// doesn't match the one of the youngest control plane machine (i.e. it's outdated).
+				cpms = cpmsActiveOutdatedBuilderOpenStack.WithNamespace(namespaceName).Build()
+				Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
+			})
+
+			It("should keep the CPMS unchanged", func() {
+				cpmsVersion := cpms.ObjectMeta.ResourceVersion
+				Consistently(komega.Object(cpms)).Should(HaveField("ObjectMeta.ResourceVersion", cpmsVersion))
+			})
+		})
+
+		Context("with state Active and up to date", func() {
+			BeforeEach(func() {
+				By("Creating an up to date and Active Control Plane Machine Set")
+				// Create an Active ControlPlaneMachineSet with a Provider Spec that
+				// doesn't match the one of the youngest control plane machine (i.e. it's up to date).
+				cpms = cpmsActiveUpToDateBuilderOpenStack.WithNamespace(namespaceName).Build()
+				Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
+			})
+
+			It("should keep the ControlPlaneMachineSet unchanged", func() {
+				cpmsVersion := cpms.ObjectMeta.ResourceVersion
+				Consistently(komega.Object(cpms)).Should(HaveField("ObjectMeta.ResourceVersion", cpmsVersion))
+			})
+
+		})
+	})
+
+	Context("when a Control Plane Machine Set exists with 3 Machine Sets", func() {
+		BeforeEach(func() {
+			By("Creating MachineSets")
+			create3MachineSets()
+			By("Creating Control Plane Machines")
+			create3CPMachines()
+		})
+
+		Context("with state Inactive and outdated", func() {
+			BeforeEach(func() {
+				By("Creating an outdated and Inactive Control Plane Machine Set")
+				// Create an Inactive ControlPlaneMachineSet with a Provider Spec that
+				// doesn't match the failure domains configured.
+				cpms = cpmsInactive5FDsBuilderOpenStack.WithNamespace(namespaceName).Build()
```

Comment:
```suggestion
				cpms = cpmsActiveOutdatedBuilderOpenStack.WithNamespace(namespaceName).Build()
```
@EmilienM, this fixed the issue you were mentioning earlier for me locally.